scrapy/addons.py

```diff
@@ -22,7 +22,7 @@ def __init__(self, crawler: Crawler) -> None:
         self.crawler: Crawler = crawler
         self.addons: list[Any] = []
 
-    def load_settings(self, settings: Settings) -> None:
+    def load_settings(self, settings: Settings | BaseSettings) -> None:
```

Comment:
What if we type it `BaseSettings` instead? (`Settings` is a subclass, and their API is identical, `Settings` only overrides `__init__`).